pkg/arvo/app/eth-watcher.hoon

```diff
@@ -509,12 +564,16 @@
     ::  maybe kill a timed-out update thread, maybe start a new one
     ::
     =^  stop-cards=(list card)  dog
-      ::  if still running beyond timeout time, kill it
+      ::  if still running beyond timeout time or reached to-block, kill it
       ::
       ?.  ?&  ?=(^ running.dog)
             ::
-              %+  gth  now.bowl
-              (add since.u.running.dog timeout-time.dog)
+            ?|  %+  gth  now.bowl
+                (add since.u.running.dog timeout-time.dog)
+                ::
+                !(has-reached-to-block dog)
+            ==
+            ::
```

Comment:
I think the change below, which prevents us from starting threads that have already "finished", is sufficient. The `number.watchdog` isn't going to change until we hear a result from the thread, which happens once the "last" invocation finishes, so we don't ever be in a position where we still have a thread running that's doing extraneous work.

tldr: pretty sure this change is not needed.

Reply:
Fair enough, the helper arm can also go then.